rclcpp_action/include/rclcpp_action/client.hpp

```diff
@@ -447,6 +459,13 @@ class Client : public ClientBase
     using GoalRequest = typename ActionT::Impl::SendGoalService::Request;
     auto goal_request = std::make_shared<GoalRequest>();
     goal_request->goal_id.uuid = this->generate_goal_id();
+    if (!add_goal_uuid(goal_request->goal_id.uuid)) {
```

Comment:
what if `rcl_action_send_goal_request` of send_goal_request fails? i think in that case, this filtering expression and parameters should be removed right away. so probably we would need to call `add_goal_uuid` in `ClientBase::send_goal_request` right after `rcl_action_send_goal_request` succeeds?

Reply:
Your suggestion seems well, but I am afraid it might not filter the very first feedback message at the beginning.
I think we can move the `add_goal_uuid` before `rcl_action_send_goal_request`, and call `remove_goal_uuid` if `rcl_action_send_goal_request` failed. 

```cpp
void
ClientBase::send_goal_request(std::shared_ptr<void> request, ResponseCallback callback)
{
  std::unique_lock<std::mutex> guard(pimpl_->goal_requests_mutex);
  int64_t sequence_number;
  <#add#> to get uuid from `request`  (a little ugly here)
  <#add#> call `add_goal_uuid`
  rcl_ret_t ret = rcl_action_send_goal_request(
    pimpl_->client_handle.get(), request.get(), &sequence_number);
  if (RCL_RET_OK != ret) {
    <#add#> call `remove_goal_uuid`
    rclcpp::exceptions::throw_from_rcl_error(ret, "failed to send goal request");
  }
  assert(pimpl_->pending_goal_responses.count(sequence_number) == 0);
  pimpl_->pending_goal_responses[sequence_number] = callback;
}
```

What do you think about that?

Reply:
> Your suggestion seems well, but I am afraid it might not filter the very first feedback message at the beginning.

this is also correct.

> What do you think about that?

it makes sense to me.